bin/servicerouter.py

```diff
@@ -547,17 +577,38 @@ def config(apps, groups):
                 "adding virtual host for app with hostname %s", app.hostname)
             cleanedUpHostname = re.sub(r'[^a-zA-Z0-9\-]', '_', app.hostname)
 
-            http_frontend_acl = templater.haproxy_http_frontend_acl
-            http_frontends += http_frontend_acl.format(
-                cleanedUpHostname=cleanedUpHostname,
-                hostname=app.hostname,
-                backend=backend
-            )
-            https_frontend_acl = templater.haproxy_https_frontend_acl
-            https_frontends += https_frontend_acl.format(
-               hostname=app.hostname,
-               backend=backend
-            )
+            if http_frontends:
+                http_frontend_acl = templater.haproxy_http_frontend_acl
+                http_frontends += http_frontend_acl.format(
+                    cleanedUpHostname=cleanedUpHostname,
+                    hostname=app.hostname,
+                    appId=app.appId,
+                    backend=backend
+                )
+            if https_frontends:
+                https_frontend_acl = templater.haproxy_https_frontend_acl
+                https_frontends += https_frontend_acl.format(
+                    cleanedUpHostname=cleanedUpHostname,
+                    hostname=app.hostname,
+                    appId=app.appId,
+                    backend=backend
+                )
+
+        if app.mode == 'http':
+            logger.debug(
+                "adding virtual host for app with id %s", app.appId)
+            cleanedUpAppId = re.sub(r'[^a-zA-Z0-9\-]', '_', app.appId)
+
+            if http_appid_frontends:
+                http_appid_frontend_acl = templater.haproxy_http_frontend_appid_acl
+                http_appid_frontends += http_appid_frontend_acl.format(
+                    cleanedUpAppId=cleanedUpAppId,
+                    hostname=app.hostname,
+                    appId=app.appId,
+                    backend=backend
+                )
+
+        if app.mode == 'http':
```

Comment:
I don't see what the value of this extra if is, couldn't this statement just be part of the previous `app.mode == 'http'`? Possibly reorder the line up a little bit higher so it isn't after the http_appid_frontends bit for readability.